### PR TITLE
Add missing parameter to assign()

### LINKED ISSUE #13: deeply nested setting
I work on a project that flings around deeply nested Python structures with wild abandon. glom nicely handles the "get something from this structure even if all the branches of the path aren't there" and now I can replace some code I wrote. Yay!

The other side of things that I need to handle is setting a value in a deeply nested structure where the branches of the path may not be there.

For example, maybe something like this which uses dicts:

```
>>> from glom import glom_set
>>> foo = {}
>>> glom_set(foo, 'a.b.c', value=5)
>>> foo
{'a': {'b': {'c': 5}}}
```

There are more complex tree manipulations that could be done, but at the moment I'm thinking about setting a single leaf value.

Is manipulating deeply nested data structures in place in-scope for glom?

### PR DESCRIPTION
**Pull request by mahmoud**
To resolve #13, this change introduces the `missing` parameter to the `Assign` spec and `assign()` convenience function. Missing data structures along the way will be backfilled by calling this callable without arguments.

```python
       >>> target = {}
       >>> assign(target, 'a.b.c', 'hi', missing=dict)
       {'a': {'b': {'c': 'hi'}}}
```

Basically tested working, pretty neat overall. It's a little bit dangerous because some corner cases can lead to partial writes, and I am vaguely concerned that the heuristic of looking for a `PathAccessError` might clobber something. But overall seems worth the risk to me! 

Sidenote: The `T`-based path made this much easier than expected, felt good!

### CODE REVIEW

**Comment by mahmoud** [comment]
Up for debate is whether to call it `missing`, `default`, or `default_factory`. I'm a fan of missing, but missing parameter sounds kinda funny.

**Comment by mahmoud** [comment]
@kurtbrose defer missing assignment to end is done now. as with many of glom's more elegant bits, it took me a while to find a configuration that hit all the cases. still, the tests are much more comprehensive now and this functionality is much more robust. Take a look!

File: glom/mutable.py [hunk 1]
```diff
@@ -105,39 +117,59 @@ def __init__(self, path, val):
...
+
+        op, arg, path = self.op, self.arg, self.path
+        try:
+            dest = scope[glom](target, self.path, scope)
+        except PathAccessError as pae:
+            if not self.missing:
+                raise
+
+            remaining_path = self._orig_path[pae.part_idx + 1:]
+            val = assign(self.missing(), remaining_path, val, missing=self.missing)
```
Comment (kurtbrose) [inline glom/mutable.py]: we can get scope etc passing through with a small tweak

```scope[glom](target, Assign(remaining_path, missing, missing=self.missing), scope)```

Comment (mahmoud) [reply glom/mutable.py]: close! but the actual line was: `val = scope[glom](self.missing(), Assign(remaining_path, val, missing=self.missing), scope)` ;)

**Pull request merged**